tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingResult.scala

```diff
@@ -5,44 +5,75 @@ import io.tofhir.engine.util.TimeUtil
 
 import java.sql.Timestamp
 
+/**
+ * Base trait representing the result of a FHIR mapping process.
+ */
+trait BaseFhirMappingResult {
+
+  /**
+   * Identifier of the job that this mapping is performed within
+   */
+  val jobId: String
+
+  /**
+   * The name of the specific mapping task being executed.
+   */
+  val mappingTaskName: String
+
+  /**
+   * Timestamp of the result
+   */
+  val timestamp: Timestamp
+
+  /**
+   * If there is a problem in the process, the JSON serialization of the source data
+   */
+  val source: String
```

Comment:
Is the "source" field always populated? Then, I suggest the update the description accordingly.